Products/ZenHub/services/DiscoverService.py

```diff
@@ -303,3 +303,25 @@ def remote_moveDevice(self, dev, path):
     def remote_getDefaultNetworks(self):
         monitor = self.dmd.Monitors.Performance._getOb(self.instance)
         return [net for net in monitor.discoveryNetworks]
+
+    @translateError
+    def remote_removeInterfaces(self, net):
+        """
+        Remove IPs for particular network
+        already assigned to interfaces (device components)
+
+        @param net - network to discover
+        @return:  a list of IPs without addresses assigned for interfaces
+        @rtype: list
+        """
+
+        full_ip_list = net.fullIpList()
+
+        for d in self.dmd.Devices.getSubDevices():
+            for interface in d.os.interfaces():
+                ip_obj = interface.getIpAddressObj()
+                if ip_obj:
+                    if net.netmask == ip_obj.netmask and ip_obj.getIp() in full_ip_list:
+                        full_ip_list.remove(ip_obj.getIp())
```

Comment:
This code does not handle the case of a network interface having multiple IP addresses.

There's an "ipaddresses" relationship on the interface that will can reference multiple addresses.  The `getIpAddressObj` method is returning whatever is the first address found in `ipaddresses`.

Something like the following will cover this situation:
```
    for addr in interface.ipaddresses():
        ip = addr.getIp()
        if net.netmask == addr.netmask and ip in full_ip_list:
            full_ip_list.remove(ip)
````